lib/bademagic_module/bluetooth/base_ble_state.dart

```diff
@@ -45,8 +45,6 @@ abstract class RetryBleState extends BleState {
         attempt++;
         if (attempt < _maxRetries) {
           logger.d("Retrying ($attempt/$_maxRetries)...");
-          await Future.delayed(
```

Comment:
We should have some delay when retrying. You can reduce the delay but Retrying would not make sense if the failure is transient and you're trying again without waiting